internal/dao/pod.go

```diff
@@ -500,19 +500,16 @@ func (p *Pod) isControlled(path string) (string, bool, error) {
 	return "", false, nil
 }
 
-func getDefaultLogContainer(po v1.Pod) string {
-	defaultContainer := po.GetAnnotations()[defaultLogContainerAnnotation]
-	if defaultContainer == "" {
-		return ""
+func getDefaultLogContainer(po v1.Pod) (string, bool) {
+	defaultContainer, ok := po.GetAnnotations()[defaultLogContainerAnnotation]
```

Comment:
Reverse the logic here ie if ok then traverse containers. Otherwise log warning if the annotation does not match ann.

Reply:
done in the last commit